### ISSUE TITLE
[Bug]: Typing indicator starts and doesn't stop intermittently (race condition)

### ISSUE THREAD

**Issue by heyitsaamir** [microsoft/teams-ai#1988]
### Language

Python

### Version

latest

### Description

If you send a feedback, using feedback loop, the the typing indicator starts up (it shouldn't, right?) and it doesn't go away even after feedback received popup comes up.

### Reproduction Steps

```bash
1. Set up a bot with feedback loop.
2. Send a message with feedback loop enabled
3. Send a feedback back
...
```

**Comment by corinagum** [comment]
@heyitsaamir I don't think this is explicitly related to feedback loop. We have confirmation that [typing indicator](https://github.com/microsoft/teams-ai/issues/1275) has weird behavior on the client side, so nothing we can do on our end. I'm following up on a few threads to confirm, but I have seen the typing indicator not go away for different samples intermittently, not just feedback loop. 

Could you confirm through the network trace that no activity at all is being sent from the bot or user during and after the feedback loop UI? One of the theories is that there's some 'invisible' activity that's activating the typing indicator but we don't have evidence yet.

**Comment by Stevenic** [comment]
I suspect that there's a race condition happening here... My suspicion is that we're in the middle of sending a message and the typing timer triggers causing us to start sending another typing update.  The typing update actually ends up getting processed by Teams after the message which leaves us in a final state of "typing".  @corinagum we should look to see when we turn off the timer. is it before or after the message is delivered? 

![image](https://github.com/user-attachments/assets/efedb2a3-f7b6-4c2b-b98c-394696f185b3)

**Comment by heyitsaamir** [comment]
> we're in the middle of sending a message

But we shouldn't be sending any messages (other than the invoke ack). Is that what you mean?

**Comment by corinagum** [comment]
@heyitsaamir We believe that the sending of the original message is interfering with how the typing activity is being handled. Teams AI code is following the typing indicator activity schema, but Teams client handles it differently. We think the race condition is related to this and how we are periodically sending a new typing indicator on long-running messages.  I discussed with stevenic in a meeting today - I will first try to see if I can generate a way to consistently repro the problem. We're hopeful that we can bugfix by modifying how the typing activity is sent in the teams channel to avoid the race condition. I will probably start this after FHL. 

Just FYI, I will be renaming the title of this bug :)

**Comment by corinagum** [comment]
Closing as resolved.
